x/reward/client/cli/tx.go

```diff
@@ -0,0 +1,227 @@
+package cli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+
+	"github.com/provenance-io/provenance/x/reward/types"
+)
+
+// Flag names and values
+const (
+	FlagTotalRewardPool         = "total-reward-pool"
+	FlagMaxRewardByAddress      = "max-reward-by-address"
+	FlagStartTime               = "start-time"
+	FlagClaimPeriods            = "claim-periods"
+	FlagClaimPeriodDays         = "claim-period-days"
+	FlagExpireDays              = "expire-days"
+	FlagQualifyingActions       = "qualifying-actions"
+	FlagMaxRolloverClaimPeriods = "max-rollover-periods"
+)
+
+func NewTxCmd() *cobra.Command {
+	txCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Aliases:                    []string{"r"},
+		Short:                      "Transaction commands for the reward module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	txCmd.AddCommand(
+		GetCmdRewardProgramAdd(),
+		GetCmdEndRewardProgram(),
+		GetCmdClaimReward(),
+	)
+
+	return txCmd
+}
+
+func GetCmdRewardProgramAdd() *cobra.Command {
+	actionsExampleJSON := "{\"qualifying_actions\":[{\"delegate\":{\"minimum_actions\":\"0\",\"maximum_actions\":\"0\",\"minimum_delegation_amount\":{\"denom\":\"nhash\",\"amount\":\"0\"},\"maximum_delegation_amount\":{\"denom\":\"nhash\",\"amount\":\"100\"},\"minimum_active_stake_percentile\":\"0.000000000000000000\",\"maximum_active_stake_percentile\":\"1.000000000000000000\"}}]}"
+	cmd := &cobra.Command{
+		Use:     "add-reward-program [title] [description]",
+		Args:    cobra.ExactArgs(2),
+		Aliases: []string{"arp"},
+		Short:   "Add a reward program",
+		Long:    strings.TrimSpace(`Add a reward program`),
+		Example: fmt.Sprintf(`$ %[1]s tx reward add-reward-program \
+		"Program Title" "A short description" \
+		--total-reward-pool 580nhash \
+		--max-reward-by-address 10nhash \
+    	--start-time 2022-05-10\
+		--claim-periods 52 \
+		--max-rollover-periods 4 \
+		--claim-period-days 7 \
+		--expire-days 14 \ 
+		--qualifying-actions %s
```

Comment:
Mixing numbered and non-numbered format identifiers can easily lead to problems.
```suggestion
		--qualifying-actions %[2]s
```

Reply:
Agreed, will change.